Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/audit/PatientDiscoveryAuditLogger.java

```diff
@@ -0,0 +1,92 @@
+ */
+package gov.hhs.fha.nhinc.patientdiscovery.audit;
+
+import gov.hhs.fha.nhinc.auditrepository.nhinc.proxy.AuditRepositoryProxy;
+import gov.hhs.fha.nhinc.auditrepository.nhinc.proxy.AuditRepositoryProxyObjectFactory;
+import gov.hhs.fha.nhinc.common.auditlog.LogEventRequestType;
+import gov.hhs.fha.nhinc.common.nhinccommon.AcknowledgementType;
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
+import gov.hhs.fha.nhinc.patientdiscovery.audit.transform.PatientDiscoveryTransforms;
+import java.util.Properties;
+import org.apache.log4j.Logger;
+import org.hl7.v3.PRPAIN201305UV02;
+import org.hl7.v3.PRPAIN201306UV02;
+
+/**
+ *
+ * @author achidamb
+ */
+public class PatientDiscoveryAuditLogger {
+
+    private static final Logger LOG = Logger.getLogger(PatientDiscoveryAuditLogger.class);
+    PatientDiscoveryTransforms patientDiscoveryTransforms = new PatientDiscoveryTransforms();
+    PatientDiscoveryAuditDataBuilder databuilder = new PatientDiscoveryAuditDataBuilder();
+
+    public void auditPatientDiscoveryRequestMessage(PRPAIN201305UV02 message, AssertionType assertion, 
+        NhinTargetSystemType target, String direction, String _interface, Boolean isRequesting, 
+        Properties webContextProperties, String serviceName) {
+        LOG.trace("---Begin PatientDiscoveryRequestAuditLogger.auditPatientDiscoveryRequestMessage()---");
+        LogEventRequestType auditLogMsg = patientDiscoveryTransforms.transformRequestToAuditMsg(message, assertion, 
+            target, direction, _interface, isRequesting, webContextProperties, serviceName, databuilder);
+        auditLogMessages(auditLogMsg, assertion);
+        LOG.trace("---End PatientDiscoveryRequestAuditLogger.auditPatientDiscoveryRequestMessage()---");
+    }
+
+    public void auditPatientDiscoveryResponseMessage(PRPAIN201306UV02 response, AssertionType assertion, 
+        NhinTargetSystemType target, String direction, String _interface, Boolean isRequesting, 
+        Properties webContextProperties, String serviceName) {
+        LOG.trace("---Begin PatientDiscoveryResponseAuditLogger.auditPatientDiscoveryResponseMessage()---");
+        LogEventRequestType auditLogMsg = patientDiscoveryTransforms.transformResponseToAuditMsg(response, assertion, 
+            target, direction, _interface, isRequesting, webContextProperties, serviceName, databuilder);
+        auditLogMessages(auditLogMsg, assertion);
+        LOG.trace("---End PatientDiscoveryResponseAuditLogger.auditPatientDiscoveryResponseMessage()---");
+    }
+
+    private void auditLogMessages(LogEventRequestType auditLogMsg, AssertionType assertion) {
+        if (auditLogMsg != null && auditLogMsg.getAuditMessage() != null) {
+            audit(auditLogMsg, assertion);
+        } else {
+            LOG.error("PatientDiscoveryAuditLogger auditLogMsg is null");
+        }
+    }
+
+    /**
```

Comment:
It's a little strange that this class has java doc for a private method and nothing for the 2 public methods.  Nitpicking.